src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,8 +563,10 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
-            
-            if (!FileSystem.Current.DirectoryExists(fullsourceDirName))
+
+            // Windows will throw if the source file/directory doesn't exist, we preemptively check
+            // to make sure our cross platform behavior matches NetFX behavior.
+            if (!FileSystem.Current.DirectoryExists(fullsourceDirName) && !FileSystem.Current.FileExists(fullsourceDirName))
```

Comment:
We dont need API approval to add `FileSystem.PathExists` right now to make it efficient, but I'm fine of course with this as is.